asg3/HashDict.cpp

```diff
@@ -1,232 +1,240 @@
-  this->table[i] = item;
-}
-
-string* HashDict::getTable() const {
-  return this->table;
-}
-
-string HashDict::getTableItem(int i) const {
-  return this->table[i];
-}
-
-void HashDict::setSize(int i) {
-  this->size = i;
-}
-int HashDict::getSize() const {
-  return this->size;
-}
-
-void HashDict::setNElements(int i) {
-  this->nElements = i;
-}
-int HashDict::getNElements() const {
-  return this->nElements;
-}
-// GETTERS/SETTERS END
-
-
-// Helper used by HashDict::nextPrime
-bool HashDict::isPrime(int size) {
-  for (int i=2; i<size; i++) { 
-    if(size % i == 0){
-      return false;
-   }
-  }
-  return true;
-}
-
-// Helper to find next prime number
-int HashDict::nextPrime(int size) {
-  while (!isPrime(size)) { 
-    size++;
-  }
-  return size;
-}
-
-// Hash function
-unsigned int HashDict::hash(string& word) {
-  unsigned int hashVal = 0;
-
-  for (int i = 0; i < word.size(); i++) {
-    hashVal = 37 * hashVal + word[i];
-  }
-
-  return hashVal % this->size; 
-}
-
-// Rehash the table/reinsert all elements
-void HashDict::rehash() {
-
-  // Get items from current table
-  int oldSize = this->size;
-  string *temp = new string[this->size];
-  for (int i = 0; i < oldSize; i++) {
-    temp[i] = this->table[i];
-  }
-
-  // Delete and recreate the class table
-  delete[] this->table;
-  this->size = this->nextPrime(this->size * 2);
-  this->table = new string[this->size];
-  this->nElements = 0;
-  
-  // Insert elements from old table to new table
-  for (int i = 0; i < oldSize; i++) {
-    if (!temp[i].empty()) {
-      this->AddEntry(temp[i]);
-    }
-  }
-  delete[] temp;
-}
-
-// When adding value, pass in empty string
-// When finding value, pass in search term
-// Returns the location of the matched string, returns -1 if not found
-int HashDict::resolveCollision(string word, int base) {
-	int i = 1;
-  int curr = (base + i) % this->size;
-	while(this->table[curr].compare(word) != 0 && !this->table[curr].empty()) {
-    // Multiple index by 2 as long as there isn't a word match and the current index isn't empty
-		i *= 2;
-    curr = (base + i) % this->size;
-	}
-	
-  // Return -1 if we did not find a search term
-	if(!word.empty() && this->table[curr].empty()) {
-		curr = -1;
-	} 
-	return curr;
-}
-
-// Add an entry
-void HashDict::AddEntry(string anEntry) {
-  unsigned int val = hash(anEntry);
-
-  // cout << "Root index: " << val << ", word: " << anEntry << endl;
-
-  if(!this->table[val].empty()) {
-    // Resolve collision using quadratic probing
-    val = this->resolveCollision("",val);
-  }
-  this->table[val] = anEntry;
-  this->nElements ++;
-
-  // Rehash if table capacity is >= 50%
-  if ((float)this->nElements / (float)this->size >= 0.5) {
-    this->rehash();
-  }
-}
-
-// Search for an item in the dictionary
-bool HashDict::FindEntry(string key) {
-  bool found = false;
-  int val = hash(key);
-  if(!table[val].empty()) {
-    if(table[val] == key) {
-      found = true;
-    }
-  }
```

Comment:
This block caused an issue where if the table cell was not empty, but did not match the key we did not attempt to resolve the collision. Resolved by moving the check for empty into the else block.